source/features/conversation-timeline-filter.tsx

```diff
@@ -0,0 +1,174 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import {observe} from 'selector-observer';
+import * as pageDetect from 'github-url-detection';
+import {CheckIcon, GearIcon, XIcon} from '@primer/octicons-react';
+
+import features from '.';
+
+enum FilterSettings {
+	ShowAll = 1,
+	ShowOnlyComments = 2,
+	ShowOnlyUnresolvedComments = 3,
+	ShowOnlyUnresolvedReviews = 4
+}
+
+let currentSettings: FilterSettings = FilterSettings.ShowAll;
+
+const showFilterName = 'rgh-show-filter';
+const menuItemCheckbox = 'rgh-filter-menu-item-checkbox';
+
+// Every element on the timeline that is recognizable by this feature will be marked with tis class.
+const timelineFiltersSelectorId = 'timeline-filters';
+
+const summaries = {
+	[FilterSettings.ShowAll]: '',
+	[FilterSettings.ShowOnlyComments]: 'Show only comments',
+	[FilterSettings.ShowOnlyUnresolvedComments]: 'Show only unresolved reviews',
+	[FilterSettings.ShowOnlyUnresolvedReviews]: 'Show unresolved comments'
```

Comment:
I previously suggested this, we don’t need enums, you can use these keys as string directly instead of 1, 2, 3, 4
```suggestion
	showOnlyUnresolvedReviews: 'Show unresolved comments'
```

If it works but typescript complains I’ll fix it

Reply:
I could not figure it out how to write it in a way that you would like.
I had 2-3 attempts and I failed miserably every time.

Reply:
test